app/src/main/java/ch/epfl/sweng/studdybuddy/activities/AddFriendsActivity.java

```diff
@@ -0,0 +1,89 @@
+import ch.epfl.sweng.studdybuddy.util.StudyBuddy;
+
+import static ch.epfl.sweng.studdybuddy.controllers.CourseSelectController.onClickAddUser;
+import static ch.epfl.sweng.studdybuddy.controllers.CourseSelectController.resetSelectViews;
+import static ch.epfl.sweng.studdybuddy.controllers.CourseSelectController.updateClickableUsers;
+import static ch.epfl.sweng.studdybuddy.util.ActivityHelper.showDropdown;
+
+public class AddFriendsActivity extends AppCompatActivity {
+    static ReferenceWrapper firebase;
+    static List<User> friendsDB = new ArrayList<>();
+    List<String> userNames = new ArrayList<>();
+    static AutoCompleteTextView autocompleteFriends;
+    public static final List<User> friendSelection = new ArrayList<>();
+    static ArrayAdapter<String> ad;
+    String uId;
+    static Button addButton;
+    private MetaGroupAdmin mga;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        firebase  = new FirebaseReference();
+        mga = new MetaGroupAdmin(firebase);
+        setContentView(R.layout.activity_find_friends);
+        setUpButtons();
+        setUpSelectedFriends();
+        setUpAutoComplete();
+        setUpDb();
+    }
+
+    private void setUpButtons() {
+        Intentable i = new Intentable(this, new Intent(this, NavigationActivity.class));
+        addButton = findViewById(R.id.btn_add);
+        User currentUser = ((StudyBuddy) AddFriendsActivity.this.getApplication()).getAuthendifiedUser();
+        uId = currentUser.getUserID().getId();
+        addButton.setOnClickListener(mga.updateFriendsOnDone(uId, friendSelection, mga, i));
+    }
+
+    private void setUpAutoComplete(){
+        ad = new ArrayAdapter<>(this,android.R.layout.simple_dropdown_item_1line, userNames);
+        autocompleteFriends = findViewById(R.id.friendsComplete);
+        autocompleteFriends.setAdapter(ad);
+        autocompleteFriends.setOnClickListener(showDropdown(autocompleteFriends));
+        InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
+        autocompleteFriends.setOnItemClickListener(onClickAddUser(friendSelection, friendsDB, resetSelectViews(addButton, autocompleteFriends, imm)));
+    }
+
+    private void setUpSelectedFriends() {
+        final RecyclerView selectedFriends = findViewById(R.id.friends_set);
+        RecyclerView.Adapter friendsAdapter = new ParticipantAdapter(friendSelection);
+        selectedFriends.setLayoutManager(new LinearLayoutManager(getBaseContext()));
+        selectedFriends.setAdapter(friendsAdapter);
+        AdapterAdapter adapterC = new RecyclerAdapterAdapter(friendsAdapter);
+        mga.addListenner(adapterC);
+        mga.addListenner(updateClickableUsers(addButton, friendSelection));
+    }
+
+    private void setUpDb() {
+        mga.addListenner(new ArrayAdapterAdapter(ad));
+        mga.fetchUser(friendsDB);
+        mga.fetchUserNames(userNames);
```

Comment:
Actually we could exclude users which are already friends, we might need to generate a third event listener which performs getBuddies and in fetchUserNames add a user to username only if it's not already a friend(by modifying the method to fetchNoneFriendsUsernames). We can keep fetchUser as is. But I say we eventually do it in other PR, due to the time constraints, agreed?

Reply:
Okay, got it, agree!